bench/bench.ml

```diff
@@ -0,0 +1,59 @@
+open Core
+open Core_bench.Std
+
+module IntDiet = Diet.Make(struct
+    type t = int
+    let compare (x: t) (y: t) = Pervasives.compare x y
+    let zero = 0
+    let succ x = x + 1
+    let pred x = x - 1
+    let add x y = x + y
+    let sub x y = x - y
+    let to_string = string_of_int
+  end)
+
+let state = Random.State.make [|994326685; 290180182; 366831641|]
```

Comment:
I just generated them using `Random.int (2^30 -1)`. The whole point is to make sure the whole benchmarks suite is deterministic and reproducible, which it won't be with `self_init`.

A comment could maybe clarify that.

Reply:
OTOH, the benchmark will run a loop, so if you use random numbers (rather than fixed ones) you'll get a better idea of the mean performance (compared to the performance on this particular input).

Reply:
I think it's not incompatible. We can generate a series of random diet but with the same PRNG state so that you have a good idea of the average perfs while still ensuring it makes sense to compare different runs of the benchmarks.

What do you think?

Reply:
If there are enough runs I don't think there's a difference, and as `core_bench` docs use `self_init` I'd use that, but as you prefer, that's not the important part of this PR.

Reply:
Ok, that sounds good to me.

It is slightly annoying that we can't use a combination of `create_indexed` and `create_with_initialization` though as my understanding is that in the first case we'll just repeatedly run equal on a randomly generated pair of diets where in the second we can generate a new pair of diets for each run.

I suggest we use `create_with_initialization` otherwise it could really raise inconsistent results across different runs.

Reply:
I pushed something. Let me know what you think!